src/annotator/anchoring/trim-range.ts

```diff
@@ -0,0 +1,218 @@
+/**
+ * From which direction to evaluate strings or nodes: from the start of a string
+ * or range seeking FORWARDS, or from the end seeking BACKWARDS.
+ */
+enum TrimDirection {
+  FORWARDS = 1,
+  BACKWARDS,
+}
+
+/**
+ * Return the offset of the nearest non-whitespace character to `baseOffset`
+ * within `text`, looking in the `direction` indicated. Return -1 if no
+ * non-whitespace character exists between `baseOffset` and the terminus of the
+ * string (start or end depending on `direction`).
+ */
+function trimTextOffset(
```

Comment:
I think the name of this function and `trimRangeContainers` could be renamed to give better cues about what they returns. `String.prototype.trim()` and `trimRange` return a new trimmed string/range respectively, whereas this function and `trimRangeContainers` return offsets. Maybe something like `findNonSpaceOffset` / `findNonSpaceOffsetInRange`?

Reply:
How about `closestNonSpaceOffset`?

Reply:
How about `closestNonSpaceOffset` and `closestNonSpacePosition`?

Reply:
Is `closestNonSpacePosition` for searching in the range or an alternative to `closestNonSpaceOffset`? Either of those sound fine for the same-node case.